graphql/src/store/prefetch.rs

```diff
@@ -26,61 +28,23 @@ lazy_static! {
     static ref ARG_ID: String = String::from("id");
 }
 
-/// Similar to the TypeCondition from graphql_parser, but with
-/// derives that make it possible to use it as the key in a HashMap
-#[derive(Clone, Debug, PartialEq, Eq, Hash)]
-enum TypeCondition {
-    Any,
-    On(q::Name),
-}
-
-impl TypeCondition {
-    /// Return a `TypeCondition` that matches when `self` and `other` match
-    /// simultaneously. If the two conditions can never match at the same time,
-    /// return `None`
-    fn and(&self, other: &Self) -> Option<Self> {
-        use TypeCondition::*;
-        match (self, other) {
-            (Any, _) => Some(other.clone()),
-            (_, Any) => Some(self.clone()),
-            (On(name1), On(name2)) if name1 == name2 => Some(self.clone()),
-            _ => None,
-        }
-    }
+#[derive(Clone, Debug)]
+struct ObjectCondition<'a>(&'a s::ObjectType);
```

Comment:
Would be good to comment that this is just an `ObjectType` that implements `Hash` and `Eq`

Reply:
I've added a comment.